File: dataadapter/src/main/java/moonbox/catalyst/jdbc/CatalystResultSet.java
package moonbox.catalyst.jdbc;

import java.io.InputStream;
import java.io.Reader;
import java.math.BigDecimal;
import java.net.URL;
import java.sql.*;
import java.util.Calendar;
import java.util.Map;

public class CatalystResultSet implements ResultSet {
  public boolean next() throws SQLException {
    return false;
  }

  public void close() throws SQLException {

  }

  public boolean wasNull() throws SQLException {
    return false;
  }

  public String getString(int columnIndex) throws SQLException {
    return null;
  }

  public boolean getBoolean(int columnIndex) throws SQLException {
    return false;
  }

  public byte getByte(int columnIndex) throws SQLException {
    return 0;
  }

  public short getShort(int columnIndex) throws SQLException {
    return 0;
  }

  public int getInt(int columnIndex) throws SQLException {
    return 0;
  }

  public long getLong(int columnIndex) throws SQLException {
    return 0;
  }

  public float getFloat(int columnIndex) throws SQLException {
    return 0;
  }

  public double getDouble(int columnIndex) throws SQLException {
    return 0;
  }

  @Deprecated
  public BigDecimal getBigDecimal(int columnIndex, int scale) throws SQLException {
    return null;
  }

  public byte[] getBytes(int columnIndex) throws SQLException {
    return new byte[0];
  }

  public Date getDate(int columnIndex) throws SQLException {
    return null;
  }

  public Time getTime(int columnIndex) throws SQLException {
    return null;
  }

  public Timestamp getTimestamp(int columnIndex) throws SQLException {
    return null;
  }

  public InputStream getAsciiStream(int columnIndex) throws SQLException {
    return null;
  }

  @Deprecated
  public InputStream getUnicodeStream(int columnIndex) throws SQLException {
    return null;
  }

  public InputStream getBinaryStream(int columnIndex) throws SQLException {
    return null;
  }

  public String getString(String columnLabel) throws SQLException {
    return null;
  }

  public boolean getBoolean(String columnLabel) throws SQLException {
    return false;
  }

  public byte getByte(String columnLabel) throws SQLException {
    return 0;
  }

  public short getShort(String columnLabel) throws SQLException {
    return 0;
  }

  public int getInt(String columnLabel) throws SQLException {
    return 0;
  }

  public long getLong(String columnLabel) throws SQLException {
    return 0;
  }

  public float getFloat(String columnLabel) throws SQLException {
    return 0;
  }

  public double getDouble(String columnLabel) throws SQLException {
    return 0;
  }

  @Deprecated
  public BigDecimal getBigDecimal(String columnLabel, int scale) throws SQLException {
    return null;
  }

  public byte[] getBytes(String columnLabel) throws SQLException {
    return new byte[0];
  }

  public Date getDate(String columnLabel) throws SQLException {
    return null;
  }

  public Time getTime(String columnLabel) throws SQLException {
    return null;
  }

  public Timestamp getTimestamp(String columnLabel) throws SQLException {
    return null;
  }

  public InputStream getAsciiStream(String columnLabel) throws SQLException {
    return null;
  }

  @Deprecated
  public InputStream getUnicodeStream(String columnLabel) throws SQLException {
    return null;
  }

  public InputStream getBinaryStream(String columnLabel) throws SQLException {
    return null;
  }

  public SQLWarning getWarnings() throws SQLException {
    return null;
  }

  public void clearWarnings() throws SQLException {

  }

  public String getCursorName() throws SQLException {
    return null;
  }

  public ResultSetMetaData getMetaData() throws SQLException {
    return null;
  }

  public Object getObject(int columnIndex) throws SQLException {
    return null;
  }

  public Object getObject(String columnLabel) throws SQLException {
    return null;
  }

  public int findColumn(String columnLabel) throws SQLException {
    return 0;
  }

  public Reader getCharacterStream(int columnIndex) throws SQLException {
    return null;
  }

  public Reader getCharacterStream(String columnLabel) throws SQLException {
    return null;
  }

  public BigDecimal getBigDecimal(int columnIndex) throws SQLException {
    return null;
  }

  public BigDecimal getBigDecimal(String columnLabel) throws SQLException {
    return null;
  }

  public boolean isBeforeFirst() throws SQLException {
    return false;
  }

  public boolean isAfterLast() throws SQLException {
    return false;
  }

  public boolean isFirst() throws SQLException {
    return false;
  }

  public boolean isLast() throws SQLException {
    return false;
  }

  public void beforeFirst() throws SQLException {

  }

  public void afterLast() throws SQLException {

  }

  public boolean first() throws SQLException {
    return false;
  }

  public boolean last() throws SQLException {
    return false;
  }

  public int getRow() throws SQLException {
    return 0;
  }

  public boolean absolute(int row) throws SQLException {
    return false;
  }

  public boolean relative(int rows) throws SQLException {
    return false;
  }

  public boolean previous() throws SQLException {
    return false;
  }

  public void setFetchDirection(int direction) throws SQLException {

  }

  public int getFetchDirection() throws SQLException {
    return 0;
  }

  public void setFetchSize(int rows) throws SQLException {

  }

  public int getFetchSize() throws SQLException {
    return 0;
  }

  public int getType() throws SQLException {
    return 0;
  }

  public int getConcurrency() throws SQLException {
    return 0;
  }

  public boolean rowUpdated() throws SQLException {
    return false;
  }

  public boolean rowInserted() throws SQLException {
    return false;
  }

  public boolean rowDeleted() throws SQLException {
    return false;
  }

  public void updateNull(int columnIndex) throws SQLException {

  }

  public void updateBoolean(int columnIndex, boolean x) throws SQLException {

  }

  public void updateByte(int columnIndex, byte x) throws SQLException {

  }

  public void updateShort(int columnIndex, short x) throws SQLException {

  }

  public void updateInt(int columnIndex, int x) throws SQLException {

  }

  public void updateLong(int columnIndex, long x) throws SQLException {

  }

  public void updateFloat(int columnIndex, float x) throws SQLException {

  }

  public void updateDouble(int columnIndex, double x) throws SQLException {

  }

  public void updateBigDecimal(int columnIndex, BigDecimal x) throws SQLException {

  }

  public void updateString(int columnIndex, String x) throws SQLException {

  }

  public void updateBytes(int columnIndex, byte[] x) throws SQLException {

  }

  public void updateDate(int columnIndex, Date x) throws SQLException {

  }

  public void updateTime(int columnIndex, Time x) throws SQLException {

  }

  public void updateTimestamp(int columnIndex, Timestamp x) throws SQLException {

  }

  public void updateAsciiStream(int columnIndex, InputStream x, int length) throws SQLException {

  }

  public void updateBinaryStream(int columnIndex, InputStream x, int length) throws SQLException {

  }

  public void updateCharacterStream(int columnIndex, Reader x, int length) throws SQLException {

  }

  public void updateObject(int columnIndex, Object x, int scaleOrLength) throws SQLException {

  }

  public void updateObject(int columnIndex, Object x) throws SQLException {

  }

  public void updateNull(String columnLabel) throws SQLException {

  }

  public void updateBoolean(String columnLabel, boolean x) throws SQLException {

  }

  public void updateByte(String columnLabel, byte x) throws SQLException {

  }

  public void updateShort(String columnLabel, short x) throws SQLException {

  }

  public void updateInt(String columnLabel, int x) throws SQLException {

  }

  public void updateLong(String columnLabel, long x) throws SQLException {

  }

  public void updateFloat(String columnLabel, float x) throws SQLException {

  }

  public void updateDouble(String columnLabel, double x) throws SQLException {

  }

  public void updateBigDecimal(String columnLabel, BigDecimal x) throws SQLException {

  }

  public void updateString(String columnLabel, String x) throws SQLException {

  }

  public void updateBytes(String columnLabel, byte[] x) throws SQLException {

  }

  public void updateDate(String columnLabel, Date x) throws SQLException {

  }

  public void updateTime(String columnLabel, Time x) throws SQLException {

  }

  public void updateTimestamp(String columnLabel, Timestamp x) throws SQLException {

  }

  public void updateAsciiStream(String columnLabel, InputStream x, int length) throws SQLException {

  }

  public void updateBinaryStream(String columnLabel, InputStream x, int length) throws SQLException {

  }

  public void updateCharacterStream(String columnLabel, Reader reader, int length) throws SQLException {

  }

  public void updateObject(String columnLabel, Object x, int scaleOrLength) throws SQLException {

  }

  public void updateObject(String columnLabel, Object x) throws SQLException {

  }

  public void insertRow() throws SQLException {

  }

  public void updateRow() throws SQLException {

  }

  public void deleteRow() throws SQLException {

  }

  public void refreshRow() throws SQLException {

  }

  public void cancelRowUpdates() throws SQLException {

  }

  public void moveToInsertRow() throws SQLException {

  }

  public void moveToCurrentRow() throws SQLException {

  }

  public Statement getStatement() throws SQLException {
    return null;
  }

  public Object getObject(int columnIndex, Map<String, Class<?>> map) throws SQLException {
    return null;
  }

  public Ref getRef(int columnIndex) throws SQLException {
    return null;
  }

  public Blob getBlob(int columnIndex) throws SQLException {
    return null;
  }

  public Clob getClob(int columnIndex) throws SQLException {
    return null;
  }

  public Array getArray(int columnIndex) throws SQLException {
    return null;
  }

  public Object getObject(String columnLabel, Map<String, Class<?>> map) throws SQLException {
    return null;
  }

  public Ref getRef(String columnLabel) throws SQLException {
    return null;
  }

  public Blob getBlob(String columnLabel) throws SQLException {
    return null;
  }

  public Clob getClob(String columnLabel) throws SQLException {
    return null;
  }

  public Array getArray(String columnLabel) throws SQLException {
    return null;
  }

  public Date getDate(int columnIndex, Calendar cal) throws SQLException {
    return null;
  }

  public Date getDate(String columnLabel, Calendar cal) throws SQLException {
    return null;
  }

  public Time getTime(int columnIndex, Calendar cal) throws SQLException {
    return null;
  }

  public Time getTime(String columnLabel, Calendar cal) throws SQLException {
    return null;
  }

  public Timestamp getTimestamp(int columnIndex, Calendar cal) throws SQLException {
    return null;
  }

  public Timestamp getTimestamp(String columnLabel, Calendar cal) throws SQLException {
    return null;
  }

  public URL getURL(int columnIndex) throws SQLException {
    return null;
  }

  public URL getURL(String columnLabel) throws SQLException {
    return null;
  }

  public void updateRef(int columnIndex, Ref x) throws SQLException {

  }

  public void updateRef(String columnLabel, Ref x) throws SQLException {

  }

  public void updateBlob(int columnIndex, Blob x) throws SQLException {

  }

  public void updateBlob(String columnLabel, Blob x) throws SQLException {

  }

  public void updateClob(int columnIndex, Clob x) throws SQLException {

  }

  public void updateClob(String columnLabel, Clob x) throws SQLException {

  }

  public void updateArray(int columnIndex, Array x) throws SQLException {

  }

  public void updateArray(String columnLabel, Array x) throws SQLException {

  }

  public RowId getRowId(int columnIndex) throws SQLException {
    return null;
  }

  public RowId getRowId(String columnLabel) throws SQLException {
    return null;
  }

  public void updateRowId(int columnIndex, RowId x) throws SQLException {

  }

  public void updateRowId(String columnLabel, RowId x) throws SQLException {

  }

  public int getHoldability() throws SQLException {
    return 0;
  }

  public boolean isClosed() throws SQLException {
    return false;
  }

  public void updateNString(int columnIndex, String nString) throws SQLException {

  }

  public void updateNString(String columnLabel, String nString) throws SQLException {

  }

  public void updateNClob(int columnIndex, NClob nClob) throws SQLException {

  }

  public void updateNClob(String columnLabel, NClob nClob) throws SQLException {

  }

  public NClob getNClob(int columnIndex) throws SQLException {
    return null;
  }

  public NClob getNClob(String columnLabel) throws SQLException {
    return null;
  }

  public SQLXML getSQLXML(int columnIndex) throws SQLException {
    return null;
  }

  public SQLXML getSQLXML(String columnLabel) throws SQLException {
    return null;
  }

  public void updateSQLXML(int columnIndex, SQLXML xmlObject) throws SQLException {

  }

  public void updateSQLXML(String columnLabel, SQLXML xmlObject) throws SQLException {

  }

  public String getNString(int columnIndex) throws SQLException {
    return null;
  }

  public String getNString(String columnLabel) throws SQLException {
    return null;
  }

  public Reader getNCharacterStream(int columnIndex) throws SQLException {
    return null;
  }

  public Reader getNCharacterStream(String columnLabel) throws SQLException {
    return null;
  }

  public void updateNCharacterStream(int columnIndex, Reader x, long length) throws SQLException {

  }

  public void updateNCharacterStream(String columnLabel, Reader reader, long length) throws SQLException {

  }

  public void updateAsciiStream(int columnIndex, InputStream x, long length) throws SQLException {

  }

  public void updateBinaryStream(int columnIndex, InputStream x, long length) throws SQLException {

  }

  public void updateCharacterStream(int columnIndex, Reader x, long length) throws SQLException {

  }

  public void updateAsciiStream(String columnLabel, InputStream x, long length) throws SQLException {

  }

  public void updateBinaryStream(String columnLabel, InputStream x, long length) throws SQLException {

  }

  public void updateCharacterStream(String columnLabel, Reader reader, long length) throws SQLException {

  }

  public void updateBlob(int columnIndex, InputStream inputStream, long length) throws SQLException {

  }

  public void updateBlob(String columnLabel, InputStream inputStream, long length) throws SQLException {

  }

  public void updateClob(int columnIndex, Reader reader, long length) throws SQLException {

  }

  public void updateClob(String columnLabel, Reader reader, long length) throws SQLException {

  }

  public void updateNClob(int columnIndex, Reader reader, long length) throws SQLException {

  }

  public void updateNClob(String columnLabel, Reader reader, long length) throws SQLException {

  }

  public void updateNCharacterStream(int columnIndex, Reader x) throws SQLException {

  }

  public void updateNCharacterStream(String columnLabel, Reader reader) throws SQLException {

  }

  public void updateAsciiStream(int columnIndex, InputStream x) throws SQLException {

  }

  public void updateBinaryStream(int columnIndex, InputStream x) throws SQLException {

  }

  public void updateCharacterStream(int columnIndex, Reader x) throws SQLException {

  }

  public void updateAsciiStream(String columnLabel, InputStream x) throws SQLException {

  }

  public void updateBinaryStream(String columnLabel, InputStream x) throws SQLException {

  }

  public void updateCharacterStream(String columnLabel, Reader reader) throws SQLException {

  }

  public void updateBlob(int columnIndex, InputStream inputStream) throws SQLException {

  }

  public void updateBlob(String columnLabel, InputStream inputStream) throws SQLException {

  }

  public void updateClob(int columnIndex, Reader reader) throws SQLException {

  }

  public void updateClob(String columnLabel, Reader reader) throws SQLException {

  }

  public void updateNClob(int columnIndex, Reader reader) throws SQLException {

  }

  public void updateNClob(String columnLabel, Reader reader) throws SQLException {

  }

  public <T> T getObject(int columnIndex, Class<T> type) throws SQLException {
    return null;
  }

  public <T> T getObject(String columnLabel, Class<T> type) throws SQLException {
    return null;
  }

  public <T> T unwrap(Class<T> iface) throws SQLException {
    return null;
  }

  public boolean isWrapperFor(Class<?> iface) throws SQLException {
    return false;
  }
}

File: catalyst/src/main/scala/moonbox/catalyst/jdbc/CatalystResultSet.scala
/*-
 * <<
 * Moonbox
 * ==
 * Copyright (C) 2016 - 2018 EDP
 * ==
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * >>
 */

package moonbox.catalyst.jdbc

import java.io.{InputStream, Reader}
import java.math.BigDecimal
import java.sql.{Blob, Clob, Date, NClob, Ref, ResultSet, RowId, SQLException, SQLXML, Time, Timestamp}
import java.util
import java.util.Calendar

class CatalystResultSet(iterator: Iterator[JdbcRow], index2SqlType: Map[Int, Int], columnLabel2Index: Map[String, Int]) extends ResultSet {
  var currentRow: JdbcRow = _
  var wasNullFlag: Boolean = false

  @throws[SQLException]
  override def next = {
    if (iterator.hasNext) {
      currentRow = iterator.next()
      true
    } else {
      false
    }
  }

  @throws[SQLException]
  override def close() = {
  }

  @throws[SQLException]
  override def wasNull = wasNullFlag

  @throws[SQLException]
  override def getString(columnIndex: Int) = currentRow.getString(columnIndex - 1)

  @throws[SQLException]
  override def getBoolean(columnIndex: Int) = currentRow.getBoolean(columnIndex - 1)

  @throws[SQLException]
  override def getByte(columnIndex: Int) = currentRow.getByte(columnIndex - 1)

  @throws[SQLException]
  override def getShort(columnIndex: Int) = currentRow.getShort(columnIndex - 1)

  @throws[SQLException]
  override def getInt(columnIndex: Int) = currentRow.getInt(columnIndex - 1)

  @throws[SQLException]
  override def getLong(columnIndex: Int) = currentRow.getLong(columnIndex - 1)

  @throws[SQLException]
  override def getFloat(columnIndex: Int) = currentRow.getFloat(columnIndex - 1)

  @throws[SQLException]
  override def getDouble(columnIndex: Int) = currentRow.getDouble(columnIndex - 1)

  @throws[SQLException]
  override def getBigDecimal(columnIndex: Int, scale: Int) = currentRow.getDecimal(columnIndex - 1).setScale(scale)

  @throws[SQLException]
  override def getBytes(columnIndex: Int) = {
    // TODO: test the correction on the type casted
    currentRow.getAs[Array[Byte]](columnIndex - 1)
  }

  @throws[SQLException]
  override def getDate(columnIndex: Int) = {
    new Date(currentRow.getLong(columnIndex - 1))
  }

  @throws[SQLException]
  override def getTime(columnIndex: Int) = {
    new Time(currentRow.getLong(columnIndex - 1))
  }

  @throws[SQLException]
  override def getTimestamp(columnIndex: Int) = {
    currentRow(columnIndex - 1).asInstanceOf[Timestamp]
  }

  @throws[SQLException]
  override def getAsciiStream(columnIndex: Int) = {
    // TODO: getAsciiStream
    null
  }

  @throws[SQLException]
  override def getUnicodeStream(columnIndex: Int) = {
    // TODO: getUnicodeStream
    null
  }

  @throws[SQLException]
  override def getBinaryStream(columnIndex: Int) = {
    // TODO: getBinaryStream
    null
  }

  @throws[SQLException]
  override def getString(columnLabel: String) = getString(columnLabel2Index(columnLabel))

  @throws[SQLException]
  override def getBoolean(columnLabel: String) = getBoolean(columnLabel2Index(columnLabel))

  @throws[SQLException]
  override def getByte(columnLabel: String) = getByte(columnLabel2Index(columnLabel))

  @throws[SQLException]
  override def getShort(columnLabel: String) = getShort(columnLabel2Index(columnLabel))

  @throws[SQLException]
  override def getInt(columnLabel: String) = getInt(columnLabel2Index(columnLabel))

  @throws[SQLException]
  override def getLong(columnLabel: String) = getLong(columnLabel2Index(columnLabel))

  @throws[SQLException]
  override def getFloat(columnLabel: String) = getFloat(columnLabel2Index(columnLabel))

  @throws[SQLException]
  override def getDouble(columnLabel: String) = getDouble(columnLabel2Index(columnLabel))

  @throws[SQLException]
  override def getBigDecimal(columnLabel: String, scale: Int) = getBigDecimal(columnLabel2Index(columnLabel), scale)

  @throws[SQLException]
  override def getBytes(columnLabel: String) = getBytes(columnLabel2Index(columnLabel))

  @throws[SQLException]
  override def getDate(columnLabel: String) = getDate(columnLabel2Index(columnLabel))

  @throws[SQLException]
  override def getTime(columnLabel: String) = getTime(columnLabel2Index(columnLabel))

  @throws[SQLException]
  override def getTimestamp(columnLabel: String) = getTimestamp(columnLabel2Index(columnLabel))

  @throws[SQLException]
  override def getAsciiStream(columnLabel: String) = getAsciiStream(columnLabel2Index(columnLabel))

  @throws[SQLException]
  override def getUnicodeStream(columnLabel: String) = getUnicodeStream(columnLabel2Index(columnLabel))

  @throws[SQLException]
  override def getBinaryStream(columnLabel: String) = getBinaryStream(columnLabel2Index(columnLabel))

  @throws[SQLException]
  override def getWarnings = {
    // TODO: getWarnings
    null
  }

  @throws[SQLException]
  override def clearWarnings() = {
    // TODO: clearWarnings
  }

  @throws[SQLException]
  override def getCursorName = null

  @throws[SQLException]
  override def getMetaData = {
    // TODO: getMetaData
    null
  }

  @throws[SQLException]
  override def getObject(columnIndex: Int) = {
    currentRow.getAs[AnyRef](columnIndex - 1)
  }

  @throws[SQLException]
  override def getObject(columnLabel: String) = getObject(columnLabel2Index(columnLabel))

  @throws[SQLException]
  override def findColumn(columnLabel: String) = columnLabel2Index(columnLabel)

  @throws[SQLException]
  override def getCharacterStream(columnIndex: Int) = {
    // TODO: getCharacterStream
    null
  }

  @throws[SQLException]
  override def getCharacterStream(columnLabel: String) = getCharacterStream(columnLabel2Index(columnLabel))

  @throws[SQLException]
  override def getBigDecimal(columnIndex: Int) = currentRow.getDecimal(columnIndex - 1)

  @throws[SQLException]
  override def getBigDecimal(columnLabel: String) = getBigDecimal(columnLabel2Index(columnLabel))

  @throws[SQLException]
  override def isBeforeFirst = {
    // TODO: isBeforeFirst
    false
  }

  @throws[SQLException]
  override def isAfterLast = {
    // TODO: isAfterLast
    false
  }

  @throws[SQLException]
  override def isFirst = {
    // TODO: isFirst
    false
  }

  @throws[SQLException]
  override def isLast = {
    // TODO: isLast
    false
  }

  @throws[SQLException]
  override def beforeFirst() = {
  }

  @throws[SQLException]
  override def afterLast() = {
  }

  @throws[SQLException]
  override def first = false

  @throws[SQLException]
  override def last = false

  @throws[SQLException]
  override def getRow = 0

  @throws[SQLException]
  override def absolute(row: Int) = false

  @throws[SQLException]
  override def relative(rows: Int) = false

  @throws[SQLException]
  override def previous = false

  @throws[SQLException]
  override def setFetchDirection(direction: Int) = {
  }

  @throws[SQLException]
  override def getFetchDirection = 0

  @throws[SQLException]
  override def setFetchSize(rows: Int) = {
  }

  @throws[SQLException]
  override def getFetchSize = 0

  // resultSet type
  @throws[SQLException]
  override def getType = ResultSet.TYPE_FORWARD_ONLY

  @throws[SQLException]
  override def getConcurrency = 0

  @throws[SQLException]
  override def rowUpdated = false

  @throws[SQLException]
  override def rowInserted = false

  @throws[SQLException]
  override def rowDeleted = false

  @throws[SQLException]
  override def updateNull(columnIndex: Int) = {
    currentRow.values(columnIndex - 1) = null
  }

  @throws[SQLException]
  override def updateBoolean(columnIndex: Int, x: Boolean) = {
    currentRow.values(columnIndex - 1) = x
  }

  @throws[SQLException]
  override def updateByte(columnIndex: Int, x: Byte) = {
    currentRow.values(columnIndex - 1) = x
  }

  @throws[SQLException]
  override def updateShort(columnIndex: Int, x: Short) = {
    currentRow.values(columnIndex - 1) = x
  }

  @throws[SQLException]
  override def updateInt(columnIndex: Int, x: Int) = {
    currentRow.values(columnIndex - 1) = x
  }

  @throws[SQLException]
  override def updateLong(columnIndex: Int, x: Long) = {
    currentRow.values(columnIndex - 1) = x
  }

  @throws[SQLException]
  override def updateFloat(columnIndex: Int, x: Float) = {
    currentRow.values(columnIndex - 1) = x
  }

  @throws[SQLException]
  override def updateDouble(columnIndex: Int, x: Double) = {
    currentRow.values(columnIndex - 1) = x
  }

  @throws[SQLException]
  override def updateBigDecimal(columnIndex: Int, x: BigDecimal) = {
    currentRow.values(columnIndex - 1) = x
  }

  @throws[SQLException]
  override def updateString(columnIndex: Int, x: String) = {
    currentRow.values(columnIndex - 1) = x
  }

  @throws[SQLException]
  override def updateBytes(columnIndex: Int, x: scala.Array[Byte]) = {
    currentRow.values(columnIndex - 1) = x
  }

  @throws[SQLException]
  override def updateDate(columnIndex: Int, x: Date) = {
    currentRow.values(columnIndex - 1) = x.getTime
  }

  @throws[SQLException]
  override def updateTime(columnIndex: Int, x: Time) = {
    currentRow.values(columnIndex - 1) = x.getTime
  }

  @throws[SQLException]
  override def updateTimestamp(columnIndex: Int, x: Timestamp) = {
    currentRow.values(columnIndex - 1) = x.getTime
  }

  @throws[SQLException]
  override def updateAsciiStream(columnIndex: Int, x: InputStream, length: Int) = {
    // TODO: updateAsciiStream
  }

  @throws[SQLException]
  override def updateBinaryStream(columnIndex: Int, x: InputStream, length: Int) = {
  }

  @throws[SQLException]
  override def updateCharacterStream(columnIndex: Int, x: Reader, length: Int) = {
  }

  @throws[SQLException]
  override def updateObject(columnIndex: Int, x: Any, scaleOrLength: Int) = {
  }

  @throws[SQLException]
  override def updateObject(columnIndex: Int, x: Any) = {
    currentRow.values(columnIndex - 1) = x
  }

  @throws[SQLException]
  override def updateNull(columnLabel: String) = {
    updateNull(columnLabel2Index(columnLabel))
  }

  @throws[SQLException]
  override def updateBoolean(columnLabel: String, x: Boolean) = {
    updateBoolean(columnLabel2Index(columnLabel), x)
  }

  @throws[SQLException]
  override def updateByte(columnLabel: String, x: Byte) = {
    updateByte(columnLabel2Index(columnLabel), x)
  }

  @throws[SQLException]
  override def updateShort(columnLabel: String, x: Short) = {
    updateShort(columnLabel2Index(columnLabel), x)
  }

  @throws[SQLException]
  override def updateInt(columnLabel: String, x: Int) = {
    updateInt(columnLabel2Index(columnLabel), x)
  }

  @throws[SQLException]
  override def updateLong(columnLabel: String, x: Long) = {
    updateLong(columnLabel2Index(columnLabel), x)
  }

  @throws[SQLException]
  override def updateFloat(columnLabel: String, x: Float) = {
    updateFloat(columnLabel2Index(columnLabel), x)
  }

  @throws[SQLException]
  override def updateDouble(columnLabel: String, x: Double) = {
    updateDouble(columnLabel2Index(columnLabel), x)
  }

  @throws[SQLException]
  override def updateBigDecimal(columnLabel: String, x: BigDecimal) = {
    updateBigDecimal(columnLabel2Index(columnLabel), x)
  }

  @throws[SQLException]
  override def updateString(columnLabel: String, x: String) = {
    updateString(columnLabel2Index(columnLabel), x)
  }

  @throws[SQLException]
  override def updateBytes(columnLabel: String, x: scala.Array[Byte]) = {
    updateBytes(columnLabel2Index(columnLabel), x)
  }

  @throws[SQLException]
  override def updateDate(columnLabel: String, x: Date) = {
    updateDate(columnLabel2Index(columnLabel), x)
  }

  @throws[SQLException]
  override def updateTime(columnLabel: String, x: Time) = {
    updateTime(columnLabel2Index(columnLabel), x)
  }

  @throws[SQLException]
  override def updateTimestamp(columnLabel: String, x: Timestamp) = {
    updateTimestamp(columnLabel2Index(columnLabel), x)
  }

  @throws[SQLException]
  override def updateAsciiStream(columnLabel: String, x: InputStream, length: Int) = {
    updateAsciiStream(columnLabel2Index(columnLabel), x, length)
  }

  @throws[SQLException]
  override def updateBinaryStream(columnLabel: String, x: InputStream, length: Int) = {
    updateBinaryStream(columnLabel2Index(columnLabel), x, length)
  }

  @throws[SQLException]
  override def updateCharacterStream(columnLabel: String, reader: Reader, length: Int) = {
    updateCharacterStream(columnLabel2Index(columnLabel), reader, length)
  }

  @throws[SQLException]
  override def updateObject(columnLabel: String, x: Any, scaleOrLength: Int) = {
    updateObject(columnLabel2Index(columnLabel), x, scaleOrLength)
  }

  @throws[SQLException]
  override def updateObject(columnLabel: String, x: Any) = {
    updateObject(columnLabel2Index(columnLabel), x)
  }

  @throws[SQLException]
  override def insertRow() = {
  }

  @throws[SQLException]
  override def updateRow() = {
  }

  @throws[SQLException]
  override def deleteRow() = {
  }

  @throws[SQLException]
  override def refreshRow() = {
  }

  @throws[SQLException]
  override def cancelRowUpdates() = {
  }

  @throws[SQLException]
  override def moveToInsertRow() = {
  }

  @throws[SQLException]
  override def moveToCurrentRow() = {
  }

  @throws[SQLException]
  override def getStatement = null

  @throws[SQLException]
  override def getObject(columnIndex: Int, map: util.Map[String, Class[_]]) = null

  @throws[SQLException]
  override def getRef(columnIndex: Int) = null

  @throws[SQLException]
  override def getBlob(columnIndex: Int) = null

  @throws[SQLException]
  override def getClob(columnIndex: Int) = null

  @throws[SQLException]
  override def getArray(columnIndex: Int) = {
    val arr = currentRow.get(columnIndex - 1).asInstanceOf[Array[Any]]
    new JdbcArray(arr)
  }

  @throws[SQLException]
  override def getObject(columnLabel: String, map: util.Map[String, Class[_]]) = null

  @throws[SQLException]
  override def getRef(columnLabel: String) = null

  @throws[SQLException]
  override def getBlob(columnLabel: String) = null

  @throws[SQLException]
  override def getClob(columnLabel: String) = null

  @throws[SQLException]
  override def getArray(columnLabel: String) = getArray(columnLabel2Index(columnLabel))

  @throws[SQLException]
  override def getDate(columnIndex: Int, cal: Calendar) = null

  @throws[SQLException]
  override def getDate(columnLabel: String, cal: Calendar) = null

  @throws[SQLException]
  override def getTime(columnIndex: Int, cal: Calendar) = null

  @throws[SQLException]
  override def getTime(columnLabel: String, cal: Calendar) = null

  @throws[SQLException]
  override def getTimestamp(columnIndex: Int, cal: Calendar) = null

  @throws[SQLException]
  override def getTimestamp(columnLabel: String, cal: Calendar) = null

  @throws[SQLException]
  override def getURL(columnIndex: Int) = null

  @throws[SQLException]
  override def getURL(columnLabel: String) = null

  @throws[SQLException]
  override def updateRef(columnIndex: Int, x: Ref) = {
  }

  @throws[SQLException]
  override def updateRef(columnLabel: String, x: Ref) = {
  }

  @throws[SQLException]
  override def updateBlob(columnIndex: Int, x: Blob) = {
  }

  @throws[SQLException]
  override def updateBlob(columnLabel: String, x: Blob) = {
  }

  @throws[SQLException]
  override def updateClob(columnIndex: Int, x: Clob) = {
  }

  @throws[SQLException]
  override def updateClob(columnLabel: String, x: Clob) = {
  }

  @throws[SQLException]
  override def updateArray(columnIndex: Int, x: java.sql.Array) = {
  }

  @throws[SQLException]
  override def updateArray(columnLabel: String, x: java.sql.Array) = {
  }

  @throws[SQLException]
  override def getRowId(columnIndex: Int) = null

  @throws[SQLException]
  override def getRowId(columnLabel: String) = null

  @throws[SQLException]
  override def updateRowId(columnIndex: Int, x: RowId) = {
  }

  @throws[SQLException]
  override def updateRowId(columnLabel: String, x: RowId) = {
  }

  @throws[SQLException]
  override def getHoldability = 0

  @throws[SQLException]
  override def isClosed = false

  @throws[SQLException]
  override def updateNString(columnIndex: Int, nString: String) = {
  }

  @throws[SQLException]
  override def updateNString(columnLabel: String, nString: String) = {
  }

  @throws[SQLException]
  override def updateNClob(columnIndex: Int, nClob: NClob) = {
  }

  @throws[SQLException]
  override def updateNClob(columnLabel: String, nClob: NClob) = {
  }

  @throws[SQLException]
  override def getNClob(columnIndex: Int) = null

  @throws[SQLException]
  override def getNClob(columnLabel: String) = null

  @throws[SQLException]
  override def getSQLXML(columnIndex: Int) = null

  @throws[SQLException]
  override def getSQLXML(columnLabel: String) = null

  @throws[SQLException]
  override def updateSQLXML(columnIndex: Int, xmlObject: SQLXML) = {
  }

  @throws[SQLException]
  override def updateSQLXML(columnLabel: String, xmlObject: SQLXML) = {
  }

  @throws[SQLException]
  override def getNString(columnIndex: Int) = null

  @throws[SQLException]
  override def getNString(columnLabel: String) = null

  @throws[SQLException]
  override def getNCharacterStream(columnIndex: Int) = null

  @throws[SQLException]
  override def getNCharacterStream(columnLabel: String) = null

  @throws[SQLException]
  override def updateNCharacterStream(columnIndex: Int, x: Reader, length: Long) = {
  }

  @throws[SQLException]
  override def updateNCharacterStream(columnLabel: String, reader: Reader, length: Long) = {
  }

  @throws[SQLException]
  override def updateAsciiStream(columnIndex: Int, x: InputStream, length: Long) = {
  }

  @throws[SQLException]
  override def updateBinaryStream(columnIndex: Int, x: InputStream, length: Long) = {
  }

  @throws[SQLException]
  override def updateCharacterStream(columnIndex: Int, x: Reader, length: Long) = {
  }

  @throws[SQLException]
  override def updateAsciiStream(columnLabel: String, x: InputStream, length: Long) = {
  }

  @throws[SQLException]
  override def updateBinaryStream(columnLabel: String, x: InputStream, length: Long) = {
  }

  @throws[SQLException]
  override def updateCharacterStream(columnLabel: String, reader: Reader, length: Long) = {
  }

  @throws[SQLException]
  override def updateBlob(columnIndex: Int, inputStream: InputStream, length: Long) = {
  }

  @throws[SQLException]
  override def updateBlob(columnLabel: String, inputStream: InputStream, length: Long) = {
  }

  @throws[SQLException]
  override def updateClob(columnIndex: Int, reader: Reader, length: Long) = {
  }

  @throws[SQLException]
  override def updateClob(columnLabel: String, reader: Reader, length: Long) = {
  }

  @throws[SQLException]
  override def updateNClob(columnIndex: Int, reader: Reader, length: Long) = {
  }

  @throws[SQLException]
  override def updateNClob(columnLabel: String, reader: Reader, length: Long) = {
  }

  @throws[SQLException]
  override def updateNCharacterStream(columnIndex: Int, x: Reader) = {
  }

  @throws[SQLException]
  override def updateNCharacterStream(columnLabel: String, reader: Reader) = {
  }

  @throws[SQLException]
  override def updateAsciiStream(columnIndex: Int, x: InputStream) = {
  }

  @throws[SQLException]
  override def updateBinaryStream(columnIndex: Int, x: InputStream) = {
  }

  @throws[SQLException]
  override def updateCharacterStream(columnIndex: Int, x: Reader) = {
  }

  @throws[SQLException]
  override def updateAsciiStream(columnLabel: String, x: InputStream) = {
  }

  @throws[SQLException]
  override def updateBinaryStream(columnLabel: String, x: InputStream) = {
  }

  @throws[SQLException]
  override def updateCharacterStream(columnLabel: String, reader: Reader) = {
  }

  @throws[SQLException]
  override def updateBlob(columnIndex: Int, inputStream: InputStream) = {
  }

  @throws[SQLException]
  override def updateBlob(columnLabel: String, inputStream: InputStream) = {
  }

  @throws[SQLException]
  override def updateClob(columnIndex: Int, reader: Reader) = {
  }

  @throws[SQLException]
  override def updateClob(columnLabel: String, reader: Reader) = {
  }

  @throws[SQLException]
  override def updateNClob(columnIndex: Int, reader: Reader) = {
  }

  @throws[SQLException]
  override def updateNClob(columnLabel: String, reader: Reader) = {
  }

  @throws[SQLException]
  override def getObject[T](columnIndex: Int, `type`: Class[T]) = null.asInstanceOf[T]

  @throws[SQLException]
  override def getObject[T](columnLabel: String, `type`: Class[T]) = null.asInstanceOf[T]

  @throws[SQLException]
  override def unwrap[T](iface: Class[T]) = null.asInstanceOf[T]

  @throws[SQLException]
  override def isWrapperFor(iface: Class[_]) = false
}
